### DISCUSSION TITLE
PUPDevice() does not work with sensorless motors

### DISCUSSION THREAD

**Discussion by vascolp** [pybricks/support#1163] [Q&A]
Hello,

PUPDevice(Port.A) does not work with sensorless devices. Tried with these on a TechnicHub:

- 1- Powered Up Medium Motor
- 2- Powered Up Train Motor

I get an `OSError: [Errno 19] ENODEV:`

Version 3.3.0b8 (also on 3.3.0b7)

I thought this is the proper way to check device type...

**Comment by laurensvalk** [comment]
This has been moved to https://github.com/pybricks/support/issues/1162 earlier today.
